### ISSUE TITLE
Remove `$legacy` parameter from `govuk-colour`

### ISSUE THREAD

**Issue by querkmachine** [alphagov/govuk-frontend#4648]
## What

Remove the `$legacy` parameter from the `govuk-colour` Sass function.

## Why

Legacy colours were removed along with other compatibility mode code in Frontend v5. Due to an oversight, we didn't formally deprecate the parameter before 5.0, so we could not remove it entirely for that release.

Instead, we kept the parameter but rendered it non-operational so that it wouldn't trigger Sass compilation errors if it was still being used. It instead returns a suppressible warning indicating that it will not work and has been deprecated.

## Who needs to work on this

- Developers

## Who needs to review this

- Developers

## Done when

```[tasklist]
### Tasks
- [ ] The `$legacy` parameter has been removed from the `govuk-colour` function
- [ ] The test for the deprecation warning has been removed
```

**Comment by domoscargin** [comment]
This was done as part of #6326 

However, we've not yet tested a v6 pre-release with the Prototype Kit, so I'm keeping this open as blocked.

**Comment by owenatgov** [comment]
Since the `$legacy` param isn't present in the kit, I'm making the executive decision to unblock this. v6 in the prototype kit is a separate issue.

**Comment by owenatgov** [comment]
We inadvertently removed this as part of https://github.com/alphagov/govuk-frontend/pull/6326. Jobs a good'n!
